### PR TITLE
fix  issues/46 and issue/40

### PR DESCRIPTION
**Pull request by pierstab**

### CODE REVIEW

File: src/metrics.cpp [hunk 1]
```diff
@@ -367,10 +367,11 @@ int printMetrics(size_t cols, bool mining)
                 if (mapBlockIndex.count(hash) > 0 &&
                         chainActive.Contains(mapBlockIndex[hash])) {
                     int height = mapBlockIndex[hash]->nHeight;
-                    CAmount subsidy = GetBlockSubsidy(height, consensusParams);
```
Comment (tarrenj) [inline src/metrics.cpp:370]: Should be inside the `// ZEN_MOD_START`: https://github.com/zcash/zcash/blob/v1.0.14/src/metrics.cpp#L329
Is this just a name change, or am I missing the bigger picture?

**Review by tarrenj — DISMISSED** [review]
Minor NIT about `// ZEN_MOD_START` location.  Do we care enough about that to require a NIT?

File: src/rpcmining.cpp [hunk 2]
```diff
@@ -909,18 +911,34 @@ UniValue getblocksubsidy(const UniValue& params, bool fHelp)
...
     }
+    CAmount superNodeFund = ForkManager::getInstance().getCommunityFundReward(nHeight,nReward, Fork::CommunityFundType::SUPERNODE);
+    if (superNodeFund > 0) {
+        minerReward -= superNodeFund;
+    }
+
+
 // ZEN_MOD_END
     UniValue result(UniValue::VOBJ);
     result.push_back(Pair("miner", ValueFromAmount(nReward)));
```
Comment (cronicc) [inline src/rpcmining.cpp]: Should be `result.push_back(Pair("miner", ValueFromAmount(minerReward)));`

Current getblocksubsidy output:
```
{
  "miner": 12.50000000,
  "community": 1.25000000,
  "securenodes": 1.25000000,
  "supernodes": 1.25000000
}
```

**Review by cronicc — CHANGES_REQUESTED** [review]

**Comment by pierstab** [comment]
closes #40 #46

**Comment by pierstab** [comment]
@cronicc done

**Review by cronicc — APPROVED** [review]

**Review by albertog78 — APPROVED** [review]

File: src/metrics.cpp (hunk 1 above)
Comment (albertog78) [reply src/metrics.cpp:370]: Comparing to master seems ok to me

**Pull request merged**